app/models/ems_refresh/save_inventory_physical_infra.rb

```diff
@@ -115,17 +115,17 @@ def save_asset_detail_inventory(parent, hash)
   end
 
   #
-  # Saves the disks information of a storage
+  # Saves the disks information of a resource
   #
-  def save_physical_disks_inventory(physical_storage, hashes)
+  def save_physical_disks_inventory(parent, hashes)
     return if hashes.nil?
 
     # Update the associated ids
     hashes.each do |h|
-      h[:physical_storage_id] = h.delete(:physical_storage).try(:[], :id)
```

Comment:
Is the intention to be able to save physical disks from both physical_storages and canisters?  I think this is a good idea and it looks like you're going that direction by changing `save_physical_disks_inventory(physical_storage, hashes)` to `save_physical_disks_inventory(parent, hashes)` but then you need to keep this foreign_key lookup not replace it with the canisters one.

Reply:
You also need to add `:physical_disks` to the `child_keys` in `save_canisters_inventory`

Reply:
> Is the intention to be able to save physical disks from both physical_storages and canisters? I think this is a good idea and it looks like you're going that direction by changing `save_physical_disks_inventory(physical_storage, hashes)` to `save_physical_disks_inventory(parent, hashes)` but then you need to keep this foreign_key lookup not replace it with the canisters one.

@agrare I only left a `canister` foreign_key because, even though we are letting it abstract enough for `physical_disks` to have any parent, currently they'll always be inside of a `physical_storage`, what will change is that sometimes it is inside a `canister` that is inside a `physical_storage`. It's a little bit confusing at first glance, but this allow us to show all disks inside a storage even if it is inside it through another component like a canister.

Reply:
> currently they'll always be inside of a `physical_storage`

That's true of the lenovo provider, but might not be true of all physical infra providers right?  Not all physical storages have canisters, in fact I think the vast majority do not no?

Reply:
> That's true of the lenovo provider, but might not be true of all physical infra providers right? Not all physical storages have canisters, in fact I think the vast majority do not no?

Yeah, that's what bothers me. Is there another way for us to access a `physical_disk` that is inside a `canister` from a storage? Because the way foreign_key is used on the other components doesn't allow us to access `physical_disks` as an attribute for both storage and canister at the same time.

Reply:
I think that's okay, from a refresh POV we don't need to be able to access both because a disk can only belong to one or the other.

Reply:
> but then you need to keep this foreign_key lookup not replace it with the canisters one.

A `physical_disk` should always belong to a `physical_storage` and saved through association, so I guess we can keep only the lookup for a canister here, so when the `canister` is present save this disk for both store and canister.

> You also need to add `:physical_disks` to the `child_keys` in `save_canisters_inventory`

We don't plan to save a `physical_disk` to a `canister` through association as explained above

> Not all physical storages have canisters, in fact I think the vast majority do not no?

Correct, but then we only save a canister to `physical_storage` when it really exists, and the `physical_disk` will always be saved to the `physical_storage` regardless.

Reply:
@felipedf I'm confused if a physical disk should belong to a physical_storage instead of a canister then why are we doing this?
We could revert this and just add a simple canister id lookup to the old method.

Reply:
The `physical_disk` will be saved to the storage through the association and saved to the canister with a simple id lookup as you said. I reverted to the old method to use physical_storage instead parent

Reply:
Okay let me know when you have the specs in the lenovo PR ready and passing with this applied locally